src/player.js

```diff
@@ -227,6 +267,90 @@ export default class Player extends FakeEventTarget {
   }
 
   /**
+   * Sets style attributes for text tracks.
+   * @param {Object} style - text styling settings
+   * @returns {void}
+   */
+  setTextStyle(style: TextStyle): void {
+    let element = Utils.Dom.getElementById(SUBTITLES_STYLE_NAME);
+    if (!element) {
+      element = Utils.Dom.createElement('style');
+      Utils.Dom.setAttribute(element, 'id', SUBTITLES_STYLE_NAME);
+      Utils.Dom.appendChild(document.head, element);
+    }
+    let sheet = element.sheet;
+
+    while (sheet.cssRules.length) {
+      sheet.deleteRule(0);
+    }
+
+    if (this._config.playback.useNativeTextTrack) {
+      sheet.insertRule(`video.${ENGINE_CLASS_NAME}::cue { ${style.toCSS()} }`, 0);
+    } else {
+      sheet.insertRule(`#${this._playerId} .${SUBTITLES_CLASS_NAME} > div > div > div { ${style.toCSS()} }`, 0);
+    }
+  }
+
+  /**
+   * update the text display settings
+   * @param {Object} settings - text cue display settings
+   * @public
+   * @returns {void}
+   */
+  setTextDisplaySettings(settings: Object): void {
+    this._textDisplaySettings = settings;
+    this._updateCueDisplaySettings();
+    for (let i = 0; i < this._activeTextCues.length; i++) {
```

Comment:
better to check `_activeTextCues` existence. i know you init as [] in  html5 `_onCueChange` but still....